scripts/ci.js

```diff
@@ -22,6 +29,63 @@ const execaOpts = {
   }
 };
 
+const isHeadCommitOfPR = async () => {
+  const octokit = new Octokit({ auth: process.env.GITHUB_TOKEN });
+  const { data } = await octokit.pulls.listCommits({
+    owner: repoOwner,
+    repo: repoName,
+    // eslint-disable-next-line camelcase
+    pull_number: process.env.CODEBUILD_SOURCE_VERSION.replace('pr/', '')
+  });
+
+  const headCommit = data[data.length - 1].sha;
+  return headCommit === process.env.CODEBUILD_RESOLVED_SOURCE_VERSION;
+};
+
+const stackReadyForDeploy = () =>
```

Comment:
I think this might read better using the promise API. Maybe something like:

```js
const stackReadyForDeploy = async () => {
  const params = {
    StackName: `sls-${name}-${tier}-${stage}`
  };

  // eslint-disable-next-line no-console
  console.log('\nChecking if CloudFormation stack is ready...');

  // First event to finish signifies done.
  const data = await Promise.race([('stackCreateComplete', 'stackUpdateComplete')]
    .map((event) => cloudformation.waitFor(event, params).promise())
  );

  // eslint-disable-next-line no-console
  console.log('CloudFormation stack is ready\n\n');
  return data;
};
```

Reply:
I've been waiting for a real-life reason to use `Promise.race`, this is 💯